AppStore/ZplayAppStore.m

```diff
@@ -0,0 +1,118 @@
+//
+//  ZplayAppStore.m
+//  Expecta
+//
+//  Created by d on 19/10/2017.
+//
+
+#import "ZplayAppStore.h"
+#import <StoreKit/StoreKit.h>
+
+#define SYSTEM_VERSION_LESS_THAN(v)                                                                                    \
+([[[UIDevice currentDevice] systemVersion] compare:v options:NSNumericSearch] == NSOrderedAscending)
+
+@interface ZplayAppStore () <SKStoreProductViewControllerDelegate>
+
+@property (nonatomic) NSNumber *itunesID;
+@property (nonatomic) NSString *iTunesLink;
+@property (nonatomic, assign) BOOL appStoreLoaded;
+
+@property (nonatomic) SKStoreProductViewController *appStore;
+@property (nonatomic, assign) BOOL isPresented;
+
+@end
+
+@implementation ZplayAppStore
+
+- (instancetype)initWithItunesID:(NSNumber *)itunesID itunesLink:(NSString *)linkUrl {
+    self = [super init];
+
+    self.itunesID = itunesID;
+    self.iTunesLink = linkUrl;
+
+    [self preloadBelowIOS11];
+    return self;
+}
+
+- (void)present {
+    if (self.appStoreLoaded) {
+        if (self.isPresented) {
+            return;
+        }
+        if ([self topMostViewController].presentedViewController) {
+            
+            return;
+        }
+        self.isPresented = YES;
+        [[self topMostViewController] presentViewController:self.appStore
+                                                      animated:YES
+                                                    completion:^{
+                                                        [self.appStoreDelegate appStoreDidAppear];
+                                                    }];
+    } else {
+        [self openAppStore:self.iTunesLink];
+    }
+}
+
+- (UIViewController *)topMostViewController {
+    UIViewController *topController = [UIApplication sharedApplication].keyWindow.rootViewController;
+    
+    while (topController.presentedViewController) {
+        topController = topController.presentedViewController;
+    }
+    
+    return topController;
+}
+
+- (void)preloadBelowIOS11 {
+    // 使用内置appstore预缓存功能在 iOS11.0 ~ 11.2.x的系统上会出现白屏现象，打开App Store链接
```

Comment:
Using the built-in AppStore pre-caching feature will appear white screen on iOS11.0 ~ 11.2.x system, please open the App Store link